sentry-android-core/src/main/java/io/sentry/android/core/AnrV2Integration.java

```diff
@@ -0,0 +1,275 @@
+      // gets killed meanwhile, we can proceed from the last reported ANR and not process the entire
+      // list again
+      Collections.reverse(exitInfos);
+      for (ApplicationExitInfo applicationExitInfo : exitInfos) {
+        if (applicationExitInfo.getReason() == ApplicationExitInfo.REASON_ANR) {
+          if (applicationExitInfo.getTimestamp() < threshold) {
+            options
+                .getLogger()
+                .log(SentryLevel.DEBUG, "ANR happened too long ago %s.", applicationExitInfo);
+            continue;
+          }
+
+          if (applicationExitInfo.getTimestamp() <= lastReportedAnr) {
+            options
+                .getLogger()
+                .log(SentryLevel.DEBUG, "ANR has already been reported %s.", applicationExitInfo);
+            continue;
+          }
+
+          reportAsSentryEvent(applicationExitInfo, false); // do not enrich past events
+        }
+      }
+    }
+
+    private void reportAsSentryEvent(
+        final @NotNull ApplicationExitInfo exitInfo, final boolean shouldEnrich) {
+      final long anrTimestamp = exitInfo.getTimestamp();
+      final Throwable anrThrowable = buildAnrThrowable(exitInfo);
+      final AnrV2Hint anrHint =
+          new AnrV2Hint(
+              options.getFlushTimeoutMillis(), options.getLogger(), anrTimestamp, shouldEnrich);
+
+      final Hint hint = HintUtils.createWithTypeCheckHint(anrHint);
+
+      final SentryEvent event = new SentryEvent(anrThrowable);
+      event.setTimestamp(DateUtils.getDateTime(anrTimestamp));
+      event.setLevel(SentryLevel.FATAL);
+
+      final @NotNull SentryId sentryId = hub.captureEvent(event, hint);
+      final boolean isEventDropped = sentryId.equals(SentryId.EMPTY_ID);
+      if (!isEventDropped) {
+        // Block until the event is flushed to disk and the last_reported_anr marker is updated
+        if (!anrHint.waitFlush()) {
+          options
+              .getLogger()
+              .log(
+                  SentryLevel.WARNING,
+                  "Timed out waiting to flush ANR event to disk. Event: %s",
+                  event.getEventId());
+        }
+      }
+    }
+
+    private @NotNull Throwable buildAnrThrowable(final @NotNull ApplicationExitInfo exitInfo) {
+      final boolean isBackground =
+          exitInfo.getImportance() != ActivityManager.RunningAppProcessInfo.IMPORTANCE_FOREGROUND;
+
+      String message = "ANR";
+      if (isBackground) {
+        message = "Background " + message;
+      }
+
+      // TODO: here we should actually parse the trace file and extract the thread dump from there
+      // and then we could properly get the main thread stracktrace and construct a proper exception
+      final ApplicationNotResponding error =
+          new ApplicationNotResponding(message, Looper.getMainLooper().getThread());
+      final Mechanism mechanism = new Mechanism();
+      mechanism.setType("ANRv2");
```

Comment:
Not sure if we should keep it as `ANRv2` or change it to `ANR`? This way we can query on looker by mechanism I think and see who's sending new ANR events. But also, it'd be possible through the integration list.

The question here is rather: do the users care about this and want to see if the ANR has been reported using new mechanism?